ecl/hqlcpp/hqlsource.cpp

```diff
@@ -5051,6 +5066,13 @@ IHqlExpression * MonitorExtractor::invertTransforms(IHqlExpression * left, IHqlE
             OwnedHqlExpr adjusted = createValue(op == no_sub ? no_add : no_sub, right->getType(), LINK(right), LINK(left->queryChild(1)));
             return invertTransforms(left->queryChild(0), adjusted);
         }
+    case no_substring:
+        {
```

Comment:
I think this can return 

return invertTransforms(left->queryChild(0), right);

I'm not sure the logic here is correct.  It is for converting

keyedField - X = searchValue

to

keyedField = searchValue + x;

I can't think of an equivalent transformation for

keyedField[1..n] = searchValue;

the closest equivalent is

keyedField between LowerBound(searchValue, n) and UpperBound(searchVAlue, n) 

That would work if we also allowed the operator to be transformed.  As it is the code is in buildKeySegmentMonitor instead.